sre-recipes/recipe.py

```diff
@@ -40,3 +42,75 @@ def verify(self):
         Verifies that the user of the recipe found the root cause
         of the breakage
         """
+
+    @abc.abstractmethod
+    def hint(self):
+        """
+        Provides a hint about the root cause of the issue
+        """
+
+    @staticmethod
+    def _run_command(command):
+        """Runs the given command and returns any output and error"""
+        process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+        output, error = process.communicate()
+        return output, error
+
+    @staticmethod
+    def _auth_cluster(cluster="APP"):
+        """ Authenticates for kubectl commands. """
+        logging.info('Authenticating cluster')
+        get_project_command = "gcloud config list --format value(core.project)"
+        project_id, error = Recipe._run_command(get_project_command)
+        project_id = project_id.decode("utf-8").replace('"', '')
+        name = "cloud-ops-sandbox"
+        if cluster == "LOADGEN":
+            name = "loadgenerator"
+        zone_command = "gcloud container clusters list --filter name:{} --project {} --format value(zone)".format(name, project_id)
+        zone, error = Recipe._run_command(zone_command)
+        zone = zone.decode("utf-8").replace('"', '')
+        print(zone)
+        auth_command = "gcloud container clusters get-credentials {} --project {} --zone {}".format(name, project_id, zone)
+        Recipe._run_command(auth_command)
+        logging.info('Cluster has been authenticated')
+
+    @staticmethod
+    def _generate_multiple_choice(prompt, choices, correct_answer):
+        """ Creates a multiple choice quiz using numeric answers and prints to terminal. Automatically polls for user response.
+        Input:
+            prompt - (string) the question asked to the user
+            choice - (list of strings) a list of responses. They will automatically be ennumerated 
+            correct_answer - (string) the correct answer - must string match with one of the entries in the choice array
+        Output:
+            No output
+        """
+        # Verify the correct exists as a choice
+        found_ans = False
+        for choice in choices:
+            if correct_answer == choice:
+                found_ans = True
+
+        if not found_ans:
+            logging.info('Correct answer not found in available choices for prompt: {}'.format(prompt))
+            return
+
+        print(prompt)
+        for index in range(1, len(choices)+1):
+            print("\t {}) {}".format(index, choices[index-1]))
```

Comment:
When using a for loop with a index variable, I think it is generally easier for people to understand it when the variable is actually used as the list's index (i.e. choices[index] instead of choices[index-1]). This also avoids having to start your loop with 1.

I would recommend:

```
for index in range(len(choices)):
  print("\t {}) {}".format(index + 1, choices[index]))

Reply:
Fixed!